### ISSUE TITLE
Don't abort regerating index on missing ports

### ISSUE THREAD

**Issue by qcybb** [jrmarino/synth#169]
Is it possible not to abort regenerating the flavor index on missing ports?

What happening (more often than not now) is that a port gets removed, but the port entry in the category MakeFile still exists.

```
$ sudo synth status
Regenerating flavor index: this may take a while ...
Scanning entire ports tree.
 progress: 9.51%              
culprit: databases/Redis-JobQueue
  Scan aborted because 'make' encounted an error in the Makefile.
  databases/Redis-JobQueue (return code = 1)
Flavor index generation failed: ports scan
```

So instead of stopping, could you just have synth confirm the directory does not exist, ignore that stale entry in the MakeFile and move on?

The issue I am having is that I set this time aside last week to do all my port updates. Now I sit here unable to do anything tonight because of this issue. Now I am forced to postpone  the updates until next week .. that is if this issue doesn't arise again.

**Comment by Duffyx** [comment]
I would even like to add if its possible to run Synth without building the index. Poudriere doesn't need this?  I like the simplicity of Synth but the 3-5 minute before starting to build is annoying me, not the end of the world though.

**Comment by jrmarino** [comment]
It's not possible.  The index became necessary when freebsd ports introduced flavors.   The project should provide the information synth generates in it's own index, but unless something has changed, they don't.
I know the wait is horrible, but the makefile-based nature of freebsd ports causes the wait.
(For what it's worth, the ravenadm program for Ravenports which is the equivalent of synth on FreeBSD ports can generate the index in less than 1.5 seconds.  If there were as many ports available it would still take less than 4 seconds, so that's what the makefiles are doing)
